src/AmalgamatorSession.ts

```diff
@@ -282,16 +291,32 @@ export class AmalgamatorSession extends LoggingDebugSession {
             });
             // then send the event for all the other stopped threads in the same child
             if (e.body.allThreadsStopped) {
-                threadMap.forEach(([childDapIndex, childId], clientId) => {
-                    if (
-                        childDapIndex === stoppedDapIndex &&
-                        childId != intiatingThreadId
-                    ) {
-                        this.sendEvent(
-                            new StoppedEvent(reason, clientId, true, false)
-                        );
+                threadMap.forEach(
```

Comment:
I really don't understand this change. Is allThreadsStopped reporting the wrong information?

There is a TODO at line 246 that bears considering in the context of this bug:

https://github.com/eclipse-cdt-cloud/cdt-amalgamator/blob/0a3cd732902ef0d364f575ec8d8f191e9743d2bd/src/AmalgamatorSession.ts#L246

Are we trying to handle events while the childdap is still in progress incorrectly?

Reply:
Hi @jonahgraham ,

Maybe my solution is wrong.
I checked it again and I realize that the root cause is at the Thread Map. The thread Map is incorrect after call to getThreadMap, the frameID of thread in thread Map can be confused together.

Example: RH850 device the ThreadMap after call to getThreadMap is Map(5) {size: 5, 1000 => (2) [0, 1], 1001 => (2) [0, 2], 1002 => (2) [0, 3], 1003 => (2) [0, 4], 1004 => (2) [1, 1]}
But the correct frameID is Map(5) {size: 5, 1000 => (2) [0, 4], 1001 => (2) [0, 3], 1002 => (2) [0, 2], 1003 => (2) [0, 1], 1004 => (2) [1, 1]}

I trial waiting the 'initialize' event in a "TODO" at line 246 but it runs incorrectly.
I try adding time delay and the Thread Map is correct, so I think maybe we need waiting an event to get the correct ThreadMap but I have no idea or solution for it.

So if you have any suggest, please let me know.
Thank you for your support!

(Sorry, I checked at work but forgot to save the code flow (e.g. MI + DAP traces). If necessary, I will send it to you as soon as possible.)